### PR TITLE
Dala 370 integrate edc update

### PR DESCRIPTION
**Pull request by ppuetsch**
# Pull Request \<Title>
`<Description here>`
## Things to do during Peer Review
Please check all boxes before the Pull Request is merged. In case you skip a box, describe in the PRs description (that means: here) why the check is skipped.
- [x] The Github Actions (including Sonarqube Gateway and Lint Checks) are green. This is enforced by Github. 
- [x] A peer-review has been executed
  - [x] The code has been manually inspected by someone who did not implement the feature
- [x] The PR actually implements what is described in the JIRA-Issue
- [x] At least one E2E Test exists testing the new feature
- [x] Documentation is updated as required
- [x] The automated deployment is updated if required
- [x] The new version is deployed to the dev server using this branch
  - [x] It's verified that this version actually is the one deployed (check actuator/info for branch name and commit id!)
  - [x] The new feature is manually used/tested/observed on dev server
  - [x] All implemented Social Logins have been tested manually in the UI
- [ ] ~~There is at least one picture for each story, which was created before coding has started~~
- [x] If any work on the UI is to be merged, those changes were also documented in the Figma 
- [x] The local Dev stack still works: execute `startDevelopmentStack.sh`, npm serve, npm teste2e and execute Cypress Tests locally

### CODE REVIEW

File: dataland-frontend/tests/e2e/utils/Cypress.ts [hunk 1]
```diff
@@ -8,6 +8,7 @@ export function doThingsInChunks<T>(
 ): Chainable<any> {
   let promise: Promise<any> = Promise.resolve();
   for (let i = 0; i < dataArray.length; i += chunkSize) {
+    console.log(`Processing chunk ${i} to ${i + chunkSize}`);
```
Comment (stewal19) [inline dataland-frontend/tests/e2e/utils/Cypress.ts:11]: Ist das ein debugging statement, welches später noch entfernt wird?

File: dataland-frontend/tests/e2e/utils/Cypress.ts [hunk 2]
```diff
@@ -9,7 +9,11 @@ export function doThingsInChunks<T>(
   let promise: Promise<any> = Promise.resolve();
   for (let i = 0; i < dataArray.length; i += chunkSize) {
     const chunk = dataArray.slice(i, i + chunkSize);
-    promise = promise.then(() => Promise.all(chunk.map((element) => processor(element))));
+    promise = promise.then(() => {
+      console.log("Chunksize: " + chunk.length);
```
Comment (stewal19) [inline dataland-frontend/tests/e2e/utils/Cypress.ts:13]: Für was genau ist das log.console hier enthalten?

Comment (ppuetsch) [reply dataland-frontend/tests/e2e/utils/Cypress.ts:13]: das kann weg.

File: dataland-backend/src/main/kotlin/org/dataland/datalandbackend/services/DataManager.kt [hunk 3]
```diff
@@ -64,6 +90,26 @@ class DataManager(
         return dataAsStorableDataSet
     }
 
+    private fun getDataFromEdcClient(dataId: String): String {
+        val dataAsString: String
+        try {
+            dataAsString = edcClient.selectDataById(dataId)
+        } catch (e: ServerException) {
+            logger.error(
+                "Error receiving selectDataById Request to Eurodat. " +
```
Comment (emanueldfine) [inline dataland-backend/src/main/kotlin/org/dataland/datalandbackend/services/DataManager.kt:99]: "receiving by EuroDaT" oder?

File: .github/workflows/CD.yaml [hunk 4]
```diff
@@ -25,6 +25,9 @@ on:
         type: string
         default: integration
 
+env:
+  GITHUB_USER: ${{ github.actor }}
+  GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (SiegfriedSobkowiak) [inline .github/workflows/CD.yaml:30]: Warum gibt es hier neue envs die es in main nicht gibt?

File: dataland-backend/src/main/kotlin/org/dataland/datalandbackend/services/DataManager.kt [hunk 5]
```diff
@@ -42,22 +47,55 @@ class DataManager(
...
+        }
+        logger.info(
+            "Stored StorableDataSet of type ${storableDataSet.dataType} for company ID ${storableDataSet.companyId}," +
+                " Company Name ${company.companyName} received ID $dataId from EuroDaT. Correlation ID: $correlationId"
+        )
+        return dataId
+    }
+
+    override fun getDataSet(dataId: String, dataType: DataType, correlationId: String): StorableDataSet {
+        getTypeNameExpectedByDataland(dataId, dataType, correlationId)
```
Comment (SiegfriedSobkowiak) [inline dataland-backend/src/main/kotlin/org/dataland/datalandbackend/services/DataManager.kt:90]: Warum wird das hier aufgerufen wenn mit dem return value nichts gemacht wird?

Comment (SiegfriedSobkowiak) [reply dataland-backend/src/main/kotlin/org/dataland/datalandbackend/services/DataManager.kt:90]: Wenn es nur um den check geht könnte man den ja direkt aufrufen

File: dataland-backend/src/main/kotlin/org/dataland/datalandbackend/services/DataManager.kt [hunk 6]
```diff
@@ -42,22 +47,55 @@ class DataManager(
...
+        )
+        val dataId: String = storeDataSet(storableDataSet, company, correlationId)
         metaDataManager.storeDataMetaInformation(company, dataId, storableDataSet.dataType)
         return dataId
     }
 
-    override fun getDataSet(dataId: String, dataType: DataType): StorableDataSet {
-        getDataMetaInformationByIdAndVerifyDataType(dataId, dataType)
-        val dataAsString = edcClient.selectDataById(dataId)
+    private fun storeDataSet(
```
Comment (SiegfriedSobkowiak) [inline dataland-backend/src/main/kotlin/org/dataland/datalandbackend/services/DataManager.kt:62]: Die Funktion ist zu lang (siehe Sonar error)

File: dataland-backend/src/main/resources/application.properties [hunk 7]
```diff
@@ -25,3 +25,7 @@ server.error.whitelabel.enabled=false
 spring.mvc.throw-exception-if-no-handler-found=true
 spring.web.resources.add-mappings=false
 dataland.expose-error-stack-trace-to-api=false
+
+logging.level.com.zaxxer.hikari=DEBUG
```
Comment (SiegfriedSobkowiak) [inline dataland-backend/src/main/resources/application.properties:29]: Soll das so bleiben?

File: dataland-backend/src/test/kotlin/org/dataland/datalandbackend/services/DataManagerTest.kt [hunk 8]
```diff
@@ -1,33 +1,153 @@
...
+    fun `check that an exception is thrown if the received data from the data storage has an unexpected type`() {
+        val storableDataSet = addCompanyAndReturnStorableEuTaxonomyDataSetForNonFinacialsForIt()
+        val storableDataSetAsString = objectMapper.writeValueAsString(storableDataSet)
+        `when`(edcClientMock.insertData(correlationId, storableDataSetAsString)).thenReturn(
+            InsertDataResponse(
+                "XXXsomeUUIDXXX"
+            )
+        )
+        val dataId = dataManager.addDataSet(storableDataSet, correlationId)
+        val unexpectedDataTypeName = "eutaxonomy-financials"
```
Comment (SiegfriedSobkowiak) [inline dataland-backend/src/test/kotlin/org/dataland/datalandbackend/services/DataManagerTest.kt:120]: Die Variable kommt nur einmal vor warum nicht direkt in die Funktion

File: dataland-backend/src/test/kotlin/org/dataland/datalandbackend/services/DataManagerTest.kt [hunk 9]
```diff
@@ -1,33 +1,153 @@
...
+        val dataId = dataManager.addDataSet(storableDataSet, correlationId)
+        `when`(edcClientMock.selectDataById(dataId, correlationId)).thenReturn("")
+        val thrown = assertThrows<ResourceNotFoundApiException> {
+            dataManager.getDataSet(dataId, DataType("eutaxonomy-non-financials"), correlationId)
+        }
+        assertEquals("No dataset with the id: XXXsomeUUIDXXX could be found in the data store.", thrown.message)
+    }
+
+    @Test
+    fun `check that an exception is thrown if the received data from the data storage has an unexpected type`() {
```
Comment (SiegfriedSobkowiak) [inline dataland-backend/src/test/kotlin/org/dataland/datalandbackend/services/DataManagerTest.kt:111]: Die Funktion ist zu lang (siehe Sonar)

File: dataland-backend/src/test/kotlin/org/dataland/datalandbackend/services/CompanyManagerTest.kt [hunk 10]
```diff
@@ -71,7 +66,7 @@ class CompanyManagerTest(
                 setOf()
             )
             assertTrue(
-                searchResponse.any { it.companyName == company.companyName },
+                searchResponse.any { it.companyInformation.companyName == company.companyName },
```
Comment (fsoffel) [inline dataland-backend/src/test/kotlin/org/dataland/datalandbackend/services/CompanyManagerTest.kt:69]: So TestDataProvider was changed in a different commit and this was a bug?

File: .github/workflows/CD.yaml (hunk 4 above)
Comment (ppuetsch) [reply .github/workflows/CD.yaml]: removed

File: dataland-backend/src/main/kotlin/org/dataland/datalandbackend/services/DataManager.kt (hunk 5 above)
Comment (ppuetsch) [reply dataland-backend/src/main/kotlin/org/dataland/datalandbackend/services/DataManager.kt]: aufgeräumt

File: dataland-backend/src/main/kotlin/org/dataland/datalandbackend/services/DataManager.kt (hunk 6 above)
Comment (ppuetsch) [reply dataland-backend/src/main/kotlin/org/dataland/datalandbackend/services/DataManager.kt:69]: gekürzt

File: dataland-backend/src/test/kotlin/org/dataland/datalandbackend/services/DataManagerTest.kt (hunk 8 above)
Comment (ppuetsch) [reply dataland-backend/src/test/kotlin/org/dataland/datalandbackend/services/DataManagerTest.kt:113]: inlined

File: dataland-backend/src/test/kotlin/org/dataland/datalandbackend/services/DataManagerTest.kt (hunk 9 above)
Comment (ppuetsch) [reply dataland-backend/src/test/kotlin/org/dataland/datalandbackend/services/DataManagerTest.kt:106]: gekürzt

File: dataland-backend/src/main/resources/application.properties (hunk 7 above)
Comment (ppuetsch) [reply dataland-backend/src/main/resources/application.properties]: das logging ist jetzt weg, der rest soll m.m.n. bleiben

**Pull request merged**